packages/xo-web/src/xo-app/settings/cloud-configs/index.js

```diff
@@ -87,90 +106,167 @@ export default decorate([
+            await editNetworkCloudConfig(networkCloudConfigToEditId, { name: networkName, template: networkTemplate })
+          }
+          reset()
+        },
       populateForm:
-        (_, { id, name, template }) =>
+        (_, { id, name, template, type }) =>
         state => ({
           ...state,
-          name,
-          cloudConfigToEditId: id,
-          template,
+          [type === 'network' ? 'networkName' : 'name']: name,
+          [type === 'network' ? 'networkCloudConfigToEditId' : 'cloudConfigToEditId']: id,
+          [type === 'network' ? 'networkTemplate' : 'template']: template,
         }),
     },
     computed: {
       formId: generateId,
       inputNameId: generateId,
       inputTemplateId: generateId,
       isInvalid: ({ name, template }) => name.trim() === '' || (template !== undefined && template.trim() === ''),
+      isNetworkValid: ({ networkName, networkTemplate }) =>
+        networkName.trim() === '' || (networkTemplate !== undefined && networkTemplate.trim() === ''),
+      userCloudConfig: (_, { cloudConfigs }) => cloudConfigs.filter(({ type }) => type === undefined),
     },
   }),
   injectState,
-  ({ state, effects, cloudConfigs }) => (
-    <Container>
-      <Col mediumSize={6}>
-        <form id={state.formId}>
-          <div className='form-group'>
-            <label htmlFor={state.inputNameId}>
-              <strong>{_('formName')}</strong>{' '}
-            </label>
-            <input
-              className='form-control'
-              id={state.inputNameId}
-              name='name'
-              onChange={effects.setInputValue}
-              type='text'
-              value={state.name}
-            />
-          </div>{' '}
-          <div className='form-group'>
-            <label htmlFor={state.inputTemplateId}>
-              <strong>{_('settingsCloudConfigTemplate')}</strong>{' '}
-            </label>{' '}
-            <AvailableTemplateVars />
-            <DebounceTextarea
-              className='form-control text-monospace'
-              id={state.inputTemplateId}
-              name='template'
-              onChange={effects.setInputValue}
-              rows={12}
-              value={defined(state.template, DEFAULT_CLOUD_CONFIG_TEMPLATE)}
-            />
-          </div>{' '}
-          {state.cloudConfigToEditId !== undefined ? (
-            <ActionButton
-              btnStyle='primary'
-              disabled={state.isInvalid}
-              form={state.formId}
-              handler={effects.editCloudConfig}
-              icon='edit'
-            >
-              {_('formEdit')}
-            </ActionButton>
-          ) : (
-            <ActionButton
-              btnStyle='success'
-              disabled={state.isInvalid}
-              form={state.formId}
-              handler={effects.createCloudConfig}
-              icon='add'
-            >
-              {_('formCreate')}
+  ({ state, effects, networkConfigs }) => (
+    <div>
```

Comment:
I prefer using <> instead. It does not add DOM node and make styling easier if we only want to return multiples elements

Reply:
Sadly `<></>` or `<React.Fragment></React.Fragment>`  are not available in our `React` version